### PR TITLE
Fix tests to check return code of error cases

### PR DESCRIPTION
**Pull request by cbourjau**
Just some updates to the tests to also check failure exit codes correctly

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -12,8 +12,8 @@ script:
   - cargo build --verbose --all --release
   - cargo test --verbose --all --release
   - cargo install --path=. --force
-  - (cd ./example_projects/simple-binary/ && bash run.sh)
-  - (cd ./example_projects/simple-library/ && bash run.sh)
+  - (cd ./example_projects/simple-binary/ && bash run.sh && cd ../../ )
+  - (cd ./example_projects/simple-library/ && bash run.sh && cd ../../ )
```
Comment (andersk) [inline .travis.yml:16]: Not necessary: the parentheses create a subshell in which the current working directory is already independent of the parent shell.

File: example_projects/simple-binary/run.sh [hunk 2]
```diff
@@ -15,5 +15,6 @@ cargo with "echo {bin} {args}" -- run -- myargs
 cargo with "echo {bin} {args}" -- test -- myargs
 
 # Some examples which should fail
-! cargo with "echo {bin}"
-cargo with "echo {bin}" -- not-a-cargo-subcommand || echo "asdfsadfadf"
+(! cargo with "echo {bin}") && true || false
+(! cargo with "echo {bin}" -- not-a-cargo-subcommand) && true || false
+(! cargo with "some-command-that-does-not-exist" -- run) && true || false
```
Comment (andersk) [inline example_projects/simple-binary/run.sh:20]: The parentheses and `&& true` aren’t needed; `! cargo with "echo {bin}" || false` would suffice.

Perhaps `! cargo with false -- run || false` should be added as a failing example, to test a command that’s successfully `exec()`ed but returns a nonzero status.

File: .travis.yml (hunk 1 above)
Comment (cbourjau) [reply .travis.yml:16]: Thats what I thought, too. But I saw some strange failures on CI. Admittedly, my shell-foo is very low!

**Comment by cbourjau** [comment]
Thanks for the feedback! I actually decided to overhaul the entire crate yesterday, including the tests. They are now run via Rust's test framework which should make testing locally and on CI transparent. If you are interested, said work is in PR #27

**Pull request closed without merging**